test/server/http/admin_test.cc

```diff
@@ -1286,8 +1309,115 @@ TEST_F(PrometheusStatsFormatterTest, UniqueMetricName) {
            {{"another_tag_name_4", "another_tag_4-value"}});
 
   Buffer::OwnedImpl response;
-  EXPECT_EQ(4UL,
-            PrometheusStatsFormatter::statsAsPrometheus(counters_, gauges_, histograms_, response));
+  auto size =
+      PrometheusStatsFormatter::statsAsPrometheus(counters_, gauges_, histograms_, response);
+  EXPECT_EQ(4UL, size);
+}
+
+TEST_F(PrometheusStatsFormatterTest, HistogramWithNoValuesAndNoTags) {
+  HistogramWrapper h1_cumulative;
+  h1_cumulative.setHistogramValues(std::vector<uint64_t>(0));
+  Stats::HistogramStatisticsImpl h1_cumulative_statistics(h1_cumulative.getHistogram());
+
+  auto histogram = std::make_shared<NiceMock<Stats::MockParentHistogram>>();
+  histogram->name_ = "histogram1";
+  histogram->used_ = true;
+  ON_CALL(*histogram, cumulativeStatistics())
+      .WillByDefault(testing::ReturnRef(h1_cumulative_statistics));
+
+  addHistogram(histogram);
+
+  Buffer::OwnedImpl response;
+  auto size =
+      PrometheusStatsFormatter::statsAsPrometheus(counters_, gauges_, histograms_, response);
+  EXPECT_EQ(1UL, size);
+
+  const std::string expected_output = R"EOF(# TYPE envoy_histogram1 histogram
+envoy_histogram1{le="0.5"} 0
+envoy_histogram1{le="1"} 0
+envoy_histogram1{le="5"} 0
+envoy_histogram1{le="10"} 0
+envoy_histogram1{le="25"} 0
+envoy_histogram1{le="50"} 0
+envoy_histogram1{le="100"} 0
+envoy_histogram1{le="250"} 0
+envoy_histogram1{le="500"} 0
+envoy_histogram1{le="1000"} 0
+envoy_histogram1{le="2500"} 0
+envoy_histogram1{le="5000"} 0
+envoy_histogram1{le="10000"} 0
+envoy_histogram1{le="30000"} 0
+envoy_histogram1{le="60000"} 0
+envoy_histogram1{le="300000"} 0
+envoy_histogram1{le="600000"} 0
+envoy_histogram1{le="1.8e+06"} 0
```

Comment:
I can't find any docs either way on whether numbers in this format "1.8e+06" are accepted in this context. It looks like they are for the value, but it's unclear whether they are for the `le` of histogram buckets. Anyone know?

Reply:
Yes they are accepted (but would be good to not expose it as such and leave it as a full integer for consistency). I’ll make the fix